lib/arborist/rebuild.js

```diff
@@ -178,6 +179,10 @@ module.exports = cls => class Builder extends cls {
       if (bin)
         await this[_checkBins](node)
       set.add(node)
+    } else if (!install && !preinstall && await isNodeGypPackage(node.path)) {
```

Comment:
Hmm, this skips the `_checkBins` if it's a nodegyp package, and skips the node-gyp default if there's bins.

Could do something like this?

```js
const isGyp = !install && !preinstall && await isNodeGypPackage(node.path)
if (bin || preinstall || install || postinstall || isGyp) {
  if (bin)
    await this[_checkBins](node)
  if (isGyp)
    node.package.scripts.install = 'node-gyp rebuild'
  set.add(node)
}
```

Reply:
Yeah, looks good.  I'll do it!

Reply:
Done.